src/frequenz/sdk/microgrid/_old_component_data.py

```diff
@@ -0,0 +1,1287 @@
+        )
+
+    @classmethod
+    @abstractmethod
+    def from_samples(cls, samples: ComponentDataSamples) -> Self:
+        """Create a new instance from a component data object."""
+
+    @classmethod
+    def _check_category(cls, component_id: ComponentId) -> None:
+        """Check if the given component_id is of the expected type.
+
+        Args:
+            component_id: Component id to check.
+
+        Raises:
+            ValueError: if the given id is unknown or has a different type.
+        """
+        # pylint: disable-next=import-outside-toplevel,cyclic-import
+        from frequenz.sdk import microgrid
+
+        components = microgrid.connection_manager.get().component_graph.components(
+            filter_by_ids={component_id}
+        )
+        if not components:
+            raise ValueError(f"Unable to find component with {component_id}")
+        if len(components) > 1:
+            raise ValueError(f"Multiple components with id {component_id}")
+        component = components.pop()
+        if component.category != cls.CATEGORY:
+            raise ValueError(
+                f"Component with {component_id} is a {component.category}, "
+                f"not a {cls.CATEGORY}."
+            )
+
+    @classmethod
+    def subscribe(
+        cls,
+        api_client: MicrogridApiClient,
+        component_id: ComponentId,
+        *,
+        buffer_size: int = DATA_STREAM_BUFFER_SIZE,
+    ) -> Receiver[Self]:
+        """Subscribe to the component data stream."""
+        cls._check_category(component_id)
+
+        def _is_valid(messages: Self | Exception) -> TypeGuard[Self]:
+            return not isinstance(messages, Exception)
+
+        receiver = api_client.receive_component_data_samples_stream(
+            component_id, cls.METRICS, buffer_size=buffer_size
+        )
+
+        return receiver.map(cls._receive_logging_errors).filter(_is_valid)
+
+    # This needs to be a classmethod because otherwise it seems like mypy can't
+    # guarantee that the Self returned by this function is the same Self in the
+    # subscribe() method.
+    @classmethod
+    def _receive_logging_errors(
+        cls, samples: ComponentDataSamples, /
+    ) -> Self | Exception:
+        try:
+            return cls.from_samples(samples)
+        except Exception as exc:  # pylint: disable=broad-except
+            _logger.exception(
+                "Error while creating %r from samples: %r", cls.__name__, samples
+            )
+            return exc
+
+
+@dataclass(kw_only=True)
+class MeterData(ComponentData):
+    """A wrapper class for holding meter data."""
+
+    # FIXME: All of this have now a default of 0.0 because this is what it was done when
+    # we used the API v0.15, as we accessed the fields without checking if the fields
+    # really existed, so the defaul protobuf value of 0.0 for floats was used.
+    # We might beed to review this and if they are not present interpret them as None
+    # instea.
```

Comment:
This one is tricky, I guess we can leave it as it is now and will have to come back to it forcibly when removing this temporary adap**ta**tion layer.

Reply:
```suggestion
    # FIXME: All of this have now a default of 0.0 because this is what it was doing when
    # we used the API v0.15, as we accessed the fields without checking if the fields
    # really existed, so the default protobuf value of 0.0 for floats was used.
    # We might need to review this and if they are not present interpret them as None
    # instead.
```

Also, it is the "adap**ta**tion" layer.

Reply:
No comment on the original comment? :D

Reply:
oops, I agree, if a field is not there, it should be None, but can be done separately.

Reply:
I left the comment in the module, we'll have to address some comments there when removing the adaptation layer module anyway.